service/topology_coordinator.hh

```diff
@@ -64,6 +64,22 @@ future<gms::inet_address> wait_for_ip(raft::server_id id, const raft_address_map
 using raft_topology_cmd_handler_type = noncopyable_function<future<raft_topology_cmd_result>(
         raft::term_t, uint64_t, const raft_topology_cmd&)>;
 
+enum class tablet_reallocation_status {
+    success,
+    no_tablets_to_allocate,
+    not_enough_nodes,
+    more_than_one_error,
+};
+struct tablet_reallocation_result {
+    using status_map = std::map<sstring, std::set<tablet_reallocation_status>>;
+    locator::tablet_map new_tablet_map;
+    status_map status; // dc -> status
+};
+
+future<tablet_reallocation_result>
+reallocate_tablets_for_new_rf(schema_ptr s, locator::token_metadata_ptr tm,
```

Comment:
I'm new here :) but this method is not called from anywhere, but tests. What's the intended usage of it?